src/args.rs

```diff
@@ -76,36 +76,43 @@ fn parse_interface(src: &str) -> Result<IpAddr, std::net::AddrParseError> {
 }
 
 /// Checks wether the auth string is valid, i.e. it follows the syntax username:password
-fn parse_auth(src: &str) -> Result<(String, String), String> {
-    let mut split = src.splitn(2, ':');
+fn parse_auth(src: &str) -> Result<auth::RequiredAuth, String> {
+    let mut split = src.splitn(3, ':');
+    let errmsg = "Invalid credentials string, expected format is username:password".to_owned();
 
     let username = match split.next() {
         Some(username) => username,
-        None => {
-            return Err(
-                "Invalid credentials string, expected format is username:password".to_owned(),
-            )
-        }
+        None => return Err(errmsg),
     };
 
-    let password = match split.next() {
+    let second_part = match split.next() {
         // This allows empty passwords, as the spec does not forbid it
         Some(password) => password,
-        None => {
-            return Err(
-                "Invalid credentials string, expected format is username:password".to_owned(),
-            )
-        }
+        None => return Err(errmsg),
     };
 
-    // To make it Windows-compatible, the password needs to be shorter than 255 characters.
-    // After 255 characters, Windows will truncate the value.
-    // As for the username, the spec does not mention a limit in length
-    if password.len() > 255 {
-        return Err("Password length cannot exceed 255 characters".to_owned());
-    }
+    let password = match split.next() {
+        Some(hash) => match second_part {
```

Comment:
You can make this into an `if let Some(hash)` because you don't need the value in the `else` part.

Reply:
You mean like [this](https://github.com/svenstaro/miniserve/pull/78/commits/8a548fd1eba41ecca91928b855eb4241352ba1d8)?

Reply:
Yep, that seems fine. Please don't make a separate PR for stuff like that, though. Just change it here in this one. Also, you should consider running clippy on your code. It should tell you stuff like this.

Reply:
OK. I'll merge the change.